### PR TITLE
Feature/gsye 441 remove accept offer bid params

### PR DESCRIPTION
**Pull request by spyrostz**
## Reason for the proposed changes

Followup to this https://github.com/gridsingularity/gsy-e/pull/1567 , this PR is removing the already_tracked and trade_rate parameters from the market class method arguments, since they are not necessary for the operation of the exchange and it will simplify the class interface. 

## Proposed changes

-

**INTEGRATION_TESTS_BRANCH**=master
**GSY_FRAMEWORK_BRANCH**=feature/GSYE-441-remove-accept-offer-bid-params

### CODE REVIEW

File: src/gsy_e/models/market/one_sided.py [hunk 1]
```diff
@@ -315,10 +318,9 @@ def accept_offer(  # pylint: disable=too-many-locals
 
         self.bc_interface.track_trade_event(self.time_slot, trade)
 
-        if already_tracked is False:
-            self._update_stats_after_trade(trade, offer)
-            log.info("%s[TRADE][OFFER] [%s] [%s] %s",
-                     self._debug_log_market_type_identifier, self.name, trade.time_slot, trade)
+        self._update_stats_after_trade(trade, offer)
+        log.info("%s[TRADE][OFFER] [%s] [%s] %s",
+                 self._debug_log_market_type_identifier, self.name, trade.time_slot, trade)
```
Comment (hannesdiedrich) [inline src/gsy_e/models/market/one_sided.py:323]: How is the logging handled now? Will we see two lines
```
[TRADE][OFFER] ...
[TRADE][BID] ...
```
for the same trade now?

File: src/gsy_e/models/market/two_sided.py [hunk 2]
```diff
@@ -255,15 +255,18 @@ def accept_bid(self, bid: Bid,
...
+                      residual=residual_bid,
                       offer_bid_trade_info=updated_bid_trade_info,
                       fee_price=fee_price, time_slot=bid.time_slot,
                       matching_requirements=trade_offer_info.matching_requirements
                       )
 
-        if already_tracked is False:
-            self._update_stats_after_trade(trade, bid, already_tracked)
+        if not offer:
+            # This is a chain trade, therefore needs to be tracked. For the trade on the market
```
Comment (hannesdiedrich) [inline src/gsy_e/models/market/two_sided.py:266]: What do you think about setting `offer=None` in the `accept_bid` call in the engine in order to make it explicit?

**Review by hannesdiedrich — COMMENTED** [review]
A few questions, LGTM in general.
I also like your proposal to rename `TradeBidOfferInfo` into `TradePricingDetails` (even in this PR)

File: src/gsy_e/models/market/two_sided.py (hunk 2 above)
Comment (spyrostz) [reply src/gsy_e/models/market/two_sided.py:266]: Thanks, fixing

File: src/gsy_e/models/market/one_sided.py (hunk 1 above)
Comment (spyrostz) [reply src/gsy_e/models/market/one_sided.py:323]: Exactly!

**Comment by spyrostz** [comment]
> A few questions, LGTM in general. I also like your proposal to rename `TradeBidOfferInfo` into `TradePricingDetails` (even in this PR)

Thanks for the comments and agreeing to the suggestion! I will open a separate PR though in order to not overcomplicate more this chain of PRs.

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Pull request merged**